### DISCUSSION TITLE
Built-In function to remove overlap matches

### DISCUSSION THREAD

**Discussion by ccsv** [explosion/spaCy#8601] [Help: Coding & Implementations] [answered]
Currently the Matcher has no easy way of removing overlap matches while preserving the output data.

Most solutions use the filter_spans function

https://stackoverflow.com/questions/63302027/how-to-avoid-double-extracting-of-overlapping-patterns-in-spacy-with-matcher 

https://stackoverflow.com/questions/63910596/spacy-matcher-dealing-with-overlapping-matches

But these solutions discard the pattern id in the Matcher. I currently wrote some code that preserves the pattern id but changes the data from a list of tuples to a dictionary and is not optimized for speed with a few for loops

https://stackoverflow.com/questions/68251333/spacy-3-0-matcher-remove-overlaps-and-preserve-the-information-for-pattern-used

Maybe add a flag that triggers running filter span on the pattern results that are grouped or have some look ahead / look behind function for OP to determine the span of tokens in the pattern before outputting the match.

**Comment by adrianeboyd** [comment, marked as answer]
Answered on StackOverflow: https://stackoverflow.com/questions/68251333/
